plugins/services/src/js/components/modals/NewCreateServiceModalForm.js

```diff
@@ -2,81 +2,200 @@ import Ace from 'react-ace';
 import classNames from 'classnames';
 import React from 'react';
 
-import Tabs from '../../../../../../src/js/components/Tabs';
+import Batch from '../../../../../../src/js/structs/Batch';
+import JSONConfigReducers from '../../reducers/JSONConfigReducers';
+import JSONParserReducers from '../../reducers/JSONParserReducers';
+import ParserUtil from '../../../../../../src/js/utils/ParserUtil';
+import ReducerUtil from '../../../../../../src/js/utils/ReducerUtil';
+import ServiceFormSection from '../forms/ServiceFormSection';
 import TabButton from '../../../../../../src/js/components/TabButton';
 import TabButtonList from '../../../../../../src/js/components/TabButtonList';
-import TabView from '../../../../../../src/js/components/TabView';
+import Tabs from '../../../../../../src/js/components/Tabs';
 import TabViewList from '../../../../../../src/js/components/TabViewList';
+import Transaction from '../../../../../../src/js/structs/Transaction';
+import Util from '../../../../../../src/js/utils/Util';
 
-const METHODS_TO_BIND = ['handleJSONChange'];
+const METHODS_TO_BIND = [
+  'handleFormChange',
+  'handleFormBlur',
+  'handleJSONBlur',
+  'handleJSONChange'
+];
+
+const SECTIONS = [
+  ServiceFormSection
+];
 
 class NewCreateServiceModalForm extends React.Component {
   constructor() {
     super(...arguments);
 
+    let batch = new Batch();
+
+    let jsonParserReducers = ParserUtil.combineParsers(JSONParserReducers);
+    let jsonConfigReducers = ReducerUtil.combineReducers(JSONConfigReducers);
+    let inputConfigReducers = ReducerUtil.combineReducers(
+      Object.assign({}, ...SECTIONS.map((item) => {
+        return item.configReducers;
+      }))
+    );
+    let validationReducers = ReducerUtil.combineReducers(
+      Object.assign({}, ...SECTIONS.map((item) => {
+        return item.validationReducers;
+      }))
+    );
+
     this.state = {
-      jsonValue: ''
+      appConfig: {},
+      batch,
+      errors: {},
+      inputConfigReducers,
```

Comment:
Do we update reducers over the time? Otherwise `state` doesn't look like the right place to me.